#**🎯Notebook: Data collection**

Autora: Vanessa

Fecha: 2026-01-21

Objetivo: Cargar y realizar inspección inicial del dataset

_______________________
📒 Contenido
1. Configuración del entorno
2. Carga de datos
3. Inspección inicial
4. Resumen estadístico
5. Conclusiones y próximos pasos


**1. Configuración del entorno**

In [8]:
import pandas as pd
import numpy as np

**2. Carga de datos**

In [9]:
df = pd.read_csv("/content/data.csv")
df.head()

cliente_id     genero  edad pais    ciudad segmento_cliente  es_mayor  \
0  0002-ORFBO   Femenino  36.0  USA  New York             PYME         0   
1  0003-MKNFE  Masculino  45.0  USA  New York      Residencial         0   
2  0004-TLHLJ  Masculino  36.0  USA  New York             PYME         0   
3  0011-IGKFF  Masculino   NaN  USA  New York             PYME         1   
4  0013-EXCHZ   Femenino  48.0  USA  New York             PYME         1   

  tiene_pareja tiene_dependientes    latitud  ...  dias_mora  \
0           Si                 Si  40.726363  ...          0   
1           No                 No  40.678022  ...          0   
2           No                 No  40.638134  ...          0   
3           Si                 No  40.683321  ...          0   
4           Si                 No  40.764546  ...          0   

  cambio_plan_reciente  downgrade_reciente visitas_app_mensual  \
0                    1                   0                  66   
1                    0                   0                  51   
2                    0                   0                   5   
3                    0                   0                   9   
4                    0                   0                  37   

  tiempo_sesion_promedio  features_nuevas_usadas  ultimo_contacto_soporte  \
0                  11.51                       3               2025-11-27   
1                  20.17                       6               2025-11-10   
2                 109.44                       5               2025-10-29   
3                  90.94                       0               2025-11-14   
4                 113.28                       8               2025-08-12   

  competidores_area  ofertas_recibidas  precio_vs_mercado  
0                 4                  0        Competitivo  
1                 1                  2               Alto  
2                 6                  2        Competitivo  
3                 2                  3        Competitivo  
4                 1                  0               Alto  

[5 rows x 67 columns]

**2. Inspección inicial**


In [4]:
df.shape

(9701, 67)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9701 entries, 0 to 9700
Data columns (total 67 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   cliente_id               9701 non-null   object 
 1   genero                   9701 non-null   object 
 2   edad                     8925 non-null   float64
 3   pais                     9701 non-null   object 
 4   ciudad                   9701 non-null   object 
 5   segmento_cliente         9701 non-null   object 
 6   es_mayor                 9701 non-null   int64  
 7   tiene_pareja             9701 non-null   object 
 8   tiene_dependientes       9701 non-null   object 
 9   latitud                  9701 non-null   float64
 10  longitud                 9701 non-null   float64
 11  borough                  9701 non-null   object 
 12  codigo_postal            9701 non-null   int64  
 13  estado                   9701 non-null   object 
 14  fecha_registro          

**4. Resumen estadístico**

In [6]:
df.describe()

edad     es_mayor      latitud     longitud  codigo_postal  \
count  8925.000000  9701.000000  9701.000000  9701.000000    9701.000000   
mean     43.971092     0.161324    40.680135   -74.113888   10832.161324   
std      12.248348     0.367848     0.151019     0.769997     514.281652   
min      18.000000     0.000000    40.112385   -77.519855   10001.000000   
25%      35.000000     0.000000    40.634279   -73.997309   10312.000000   
50%      44.000000     0.000000    40.705993   -73.942995   11114.000000   
75%      53.000000     0.000000    40.758109   -73.877799   11239.000000   
max      74.000000     1.000000    40.910207   -73.701843   11696.000000   

       ingreso_mediano  densidad_poblacional   antiguedad  cargo_mensual  \
count      9701.000000           9701.000000  9701.000000    9701.000000   
mean      66460.674157          14736.604474    32.898876      89.254500   
std       14676.147445           7302.961070    25.605834      49.153898   
min       42000.000000           3200.000000     0.000000      15.000000   
25%       63000.000000           8500.000000     9.000000      48.280000   
50%       72000.000000          13000.000000    29.000000      90.620000   
75%       82000.000000          14000.000000    56.000000     127.660000   
max       85000.000000          28000.000000   179.000000     342.320124   

       ingresos_totales  ...  cancelacion  intentos_cobro_fallidos  \
count       9701.000000  ...  9701.000000              9701.000000   
mean        3208.115742  ...     0.156891                 0.996701   
std         3469.562390  ...     0.363717                 1.067799   
min            0.000000  ...     0.000000                 0.000000   
25%          488.040000  ...     0.000000                 0.000000   
50%         1705.550000  ...     0.000000                 1.000000   
75%         5155.840000  ...     0.000000                 2.000000   
max        29237.580000  ...     1.000000                 3.000000   

         dias_mora  cambio_plan_reciente  downgrade_reciente  \
count  9701.000000           9701.000000         9701.000000   
mean      7.632203              0.256365            0.213586   
std      10.032225              0.436648            0.409859   
min       0.000000              0.000000            0.000000   
25%       0.000000              0.000000            0.000000   
50%       5.000000              0.000000            0.000000   
75%      15.000000              1.000000            0.000000   
max      30.000000              1.000000            1.000000   

       visitas_app_mensual  tiempo_sesion_promedio  features_nuevas_usadas  \
count          9701.000000             9701.000000             9701.000000   
mean             49.405731               62.678872                4.984641   
std              28.884164               33.236751                3.153246   
min               0.000000                5.010000                0.000000   
25%              24.000000               34.230000                2.000000   
50%              50.000000               62.360000                5.000000   
75%              74.000000               91.690000                8.000000   
max              99.000000              119.990000               10.000000   

       competidores_area  ofertas_recibidas  
count        9701.000000        9701.000000  
mean            4.503762           0.981858  
std             2.284261           1.065524  
min             1.000000           0.000000  
25%             2.000000           0.000000  
50%             5.000000           1.000000  
75%             6.000000           2.000000  
max             8.000000           3.000000  

[8 rows x 36 columns]

**5. Conclusiones y próximos pasos**

El conjunto de datos contiene 9,701 registros y 67 variables, lo que indica un volumen de información suficiente para entrenar modelos de machine learning con buen nivel de generalización.

Las variables están compuestas por una mezcla de tipos de datos: 31 variables categóricas (object), 19 variables enteras (int64), 17 variables numéricas continuas (float64).

Esto sugiere que será necesario aplicar técnicas de codificación de variables categóricas (como One-Hot Encoding o Target Encoding) antes de entrenar modelos que no acepten texto.

Respecto a la calidad de los datos, se tiene que la mayoría de las columnas presentan valores completos, sin embargo, se identifican algunas variables con datos faltantes:`edad`, `dias_ultima_conexion`,
`respuesta_encuesta` y `referencias_hechas`.

Aunque el porcentaje de valores faltantes no es crítico, será necesario definir una estrategia de imputación o eliminación con el fin de evitar sesgos o pérdida de información.

Dado el alto número de variables presentes en el dataset (67 columnas), es probable que exista redundancia de información entre algunas de ellas, especialmente entre variables financieras, de uso del servicio y de interacción con soporte. Esta situación puede generar multicolinealidad, donde dos o más variables están altamente correlacionadas, lo que puede afectar negativamente el desempeño e interpretabilidad de ciertos modelos, en particular los modelos lineales.


Próximos pasos:
* Notebook 02: Análisis de calidad de datos
* Notebook 03: Limpieza de datos
* Notebook 04: Transformación de datos